Dockerfile

```
@@ -445,6 +436,18 @@ function common::show_pip_version_and_location() {
    echo "pip on path: $(which pip)"
    echo "Using pip: $(pip --version)"
 }
+
+function common::install_pip_version() {
+    echo
+    echo "${COLOR_BLUE}Installing pip version ${AIRFLOW_PIP_VERSION}${COLOR_RESET}"
+    echo
+    if [[ ${AIRFLOW_PIP_VERSION} =~ .*https.* ]]; then
+        pip install --disable-pip-version-check --no-cache-dir ${AIRFLOW_PIP_VERSION}
```

```suggestion
        pip install --disable-pip-version-check --no-cache-dir "pip @ ${AIRFLOW_PIP_VERSION}"
```

Slightly more error-proof.

Otherwise looks fine to me.

---

Right: https://peps.python.org/pep-0440/#direct-references TIL (also good that PEP explicitly mention hashes and warnings if the hash is not present).

---

Resolved.